src/index.ts

```diff
@@ -15,78 +18,108 @@ export function isType<P>(
-  (payload?: P, meta?: Object): Action<P>;
+  (payload: P, meta?: Object): Action<P>;
+}
+
+export interface ActionCreatorWithoutPayload {
+  type: string;
+  (payload?: undefined, meta?: Object): Action<undefined>;
+}
+
+
+export interface DonePayload<P, R> {
+  params: P;
+  result: R;
+}
+
+export interface FailedPayload<P, E> {
+  params: P;
+  error: E;
 }
 
-export interface AsyncActionCreators<P, R> {
+export interface AsyncActionCreators<P, R, E> {
   type: string;
   started: ActionCreator<P>;
-  done: ActionCreator<{
-    params: P;
-    result: R;
-  }>;
-  failed: ActionCreator<{
-    params: P;
-    error: any;
-  }>;
+  done: ActionCreator<DonePayload<P, R>>;
+  failed: ActionCreator<FailedPayload<P, E>>;
 }
 
 export interface ActionCreatorFactory {
-  <P>(type: string, commonMeta?: Object, error?: boolean): ActionCreator<P>;
-
-  async<P, S>(type: string, commonMeta?: Object): AsyncActionCreators<P, S>;
+  (type: string): ActionCreatorWithoutPayload;
+  <P>(type: string, commonMeta?: Object): ActionCreator<P>;
+  <P>(type: string, commonMeta: Object | undefined, isError: boolean): ActionCreator<P>;
+  <P, E>(type: string, commonMeta: Object | undefined, isError: (payload: P | E) => boolean): ActionCreator<P | E>;
+  <P, E extends Error>(type: string, commonMeta?: Object): ActionCreator<P | E>;
+  
+
+  async<P, S, E>(type: string, commonMeta?: Object): AsyncActionCreators<P, S, E>;
 }
 
+function wrapIsError(default_: ((payload: any) => boolean), isError?: ((payload: any) => boolean) | boolean) {
+  if (isError === undefined) {
+    return default_
+  }
+  if (typeof isError === "boolean") {
+    return () => isError
+  }
+  return isError
+}
 
-export default function actionCreatorFactory(prefix?: string):
+export function actionCreatorFactory(prefix?: string, defaultIsError?: ((payload: any) => boolean) | boolean):
 ActionCreatorFactory {
+
   const actionTypes = {};
+  const base = prefix ? `${prefix}/` : ""
 
-  function actionCreator<P>(type: string, commonMeta?: Object,
-                            error?: boolean): ActionCreator<P> {
-    if (actionTypes[type])
-      throw new Error(`Duplicate action type: ${type}`);
+  const defaultIsError_ = wrapIsError(p => typeof p === "Error", isError)
 
-    actionTypes[type] = true;
+  function baseActionCreator<P>(isError: (payload: P) => boolean, type: string, commonMeta?: Object): ActionCreator<P> {
+                      
+    const fullType = `${base}${type}`
 
-    const fullType = prefix ? `${prefix}/${type}` : type;
+    if (isDev) {
+      if (actionTypes[fullType])
+        throw new Error(`Duplicate action type: ${fullType}`);
+
+      actionTypes[fullType] = true;
+    }
 
     return Object.assign(
-      (payload?: P, meta?: Object) => {
-        const action: Action<P> = {
+      (payload: P, meta?: Object) => {
+        return {
           type: fullType,
           payload,
           meta: Object.assign({}, commonMeta, meta),
+          error: isError(payload)
```

Comment:
I like it more when `error` field is completely omitted for non-error actions. Most of them are not, so let's not clutter people's DevTools :)